src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java

```diff
@@ -29,62 +29,59 @@
-        validate(offenderNo, accountCodeOpl, fromDateOpl, toDateOpl);
+                                                                     final Optional<LocalDate> fromDate,
+                                                                     final Optional<LocalDate> toDate,
+                                                                     final String transactionType) {
+        validate(offenderNo, accountCodeOpl, fromDate, toDate);
 
-        var fromDate = fromDateOpl.orElse(LocalDate.now());
-        var toDate = toDateOpl.orElse(LocalDate.now());
-        checkDateRange(fromDate, toDate);
-
-        if (accountCodeOpl.isPresent()) {
-            boolean isAccountCodeExists = AccountCode.exists(accountCodeOpl.get());
-            checkState(isAccountCodeExists, "Unknown account-code " + accountCodeOpl.get());
-        }
-
-        return accountCodeOpl.map(AccountCode::byCodeName)
-            .filter(Optional::isPresent)
+        final var accountCode = accountCodeOpl
+            .map(AccountCode::byCodeName)
             .map(optionalCode -> optionalCode.get().code)
-            .map(code -> historyRepository.findForGivenAccountType(offenderNo, code, fromDate, toDate))
-            .orElse(historyRepository.findForAllAccountTypes(offenderNo, fromDate, toDate))
+            .orElse(null);
+
+        return historyRepository.getTransactionHistory(offenderNo, accountCode, fromDate.orElse(null), toDate.orElse(null), transactionType)
             .stream()
             .sorted(SORT_BY_MOST_RECENT_DATE_THEN_BY_LATEST_SEQ)
             .map(this::transform)
             .collect(Collectors.toList());
     }
 
-    private void checkDateRange(LocalDate fromDate, LocalDate toDate) {
-        var now = LocalDate.now();
-        checkState(fromDate.isBefore(toDate) || fromDate.isEqual(toDate), "toDate can't be before fromDate");
-        checkState(fromDate.isBefore(now) || fromDate.isEqual(now), "fromDate can't be in the future");
-        checkState(toDate.isBefore(now) || toDate.isEqual(now), "toDate can't be in the future");
-    }
-
     private void validate(final String offenderNo,
                           final Optional<String> accountCodeOpl,
-                          final Optional<LocalDate> fromDateOpl,
-                          final Optional<LocalDate> toDateOpl) {
+                          final Optional<LocalDate> fromDate,
+                          final Optional<LocalDate> toDate) {
 
         checkNotNull(offenderNo, "offenderNo can't be null");
         checkNotNull(accountCodeOpl, "accountCode optional can't be null");
-        checkNotNull(fromDateOpl, "fromDate optional can't be null");
-        checkNotNull(toDateOpl, "toDate optional can't be null");
+        checkNotNull(fromDate, "fromDate optional can't be null");
+        checkNotNull(toDate, "toDate optional can't be null");
+
+        if (fromDate.isPresent() && toDate.isPresent()) checkDateRange(fromDate.get(), toDate.get());
```

Comment:
I think I'm missing something here, but won't the checkNotNull throw an exception before you get to this line, so no need to check `isPresent`?  Also, not really sure why these parameters are nullable in the API only to then validate them?

Reply:
You're absolutely right! Good spot! I'll fix now.

Reply:
Oh sorry, Optional vs null - two different things. :facepalm: